common/lib/capa/capa/responsetypes.py

```diff
@@ -252,6 +256,59 @@ def evaluate_answers(self, student_answers, old_cmap):
         # log.debug('new_cmap = %s' % new_cmap)
         return new_cmap
 
+    def make_hint_div(self, hint_node, correct, label=None, hint_text=''):
+        """
+        Given the xml extended hint node and the boolean correctness, returns the extended hint div,
+        like this:
+          <div class="feedback_hint_incorrect">Incorrect: the answer is not 42</div>
+        Optionally, the hint label and the hint_text to use may be passed in,
+        overriding the text found in the hint node, which can be passed as None.
+        This function supports many argument combinations in order to work with the many
+        different contexts where extended hints are constructed.
+        A label with the value '' means inhibit all labeling, while the None means
+        no specific label is provided and we should get the default.
+        The empty string is returned, meaning no hint, if after processing all the arguments,
+        there is no hint.
```

Comment:
I think I would prefer to see this docstring conform a bit more to [pep257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings), that is, a one line brief summary, a blank line, and then a more elaborate description. It owuld be good if the elaborate description used the args, eg:
```python
"""
Brief one-line summary.

Keyword args:
    * hint_node: What this is (type) and what it does
    * correct: What this is (type) and what it does
    * label: String. A label with the value '' means inhibit all labeling, while None means
        no specific label is provided and we should get the default.
    * hint_text: What this is and what it does

Returns: a more elaborate description of what this returns than the one-line summary gives, if necessary.
"""
```